pontoon/sync/tests/test_tasks.py

```diff
@@ -150,3 +154,65 @@ def test_remove_duplicate_approvals(self):
         duplicate_translation.refresh_from_db()
         assert_true(duplicate_translation.approved)
         assert_equal(duplicate_translation.approved_date, aware_datetime(1970, 1, 3))
+
+
+class UserError(Exception):
+    pass
+
+class SyncExecutionTests(TestCase):
+    def setUp(self):
+        self.project = ProjectFactory.create()
+
+    def test_serial_task(self):
+        """
+        Test if sync will create lock in cache and release this after task is done.
+        """
+        @serial_task(100)
+        def test_task(self, callback):
+            return callback()
+
+        def execute_second_inner_task():
+            return test_task.delay(lambda: None)
+
+        first_call = test_task.delay(execute_second_inner_task)
+        second_call = first_call.get()
+
+        assert_true(first_call.successful())
+        assert_true(second_call.failed())
+        assert_raises(RuntimeError, second_call.get)
+
+    def test_release_lock_after_timeout(self):
+        """
+        Tests if lock is released after specified timeout.
+        """
+        @serial_task(3)
+        def timeout_task(self):
+            return wait(lambda: not cache.get('serial_task.pontoon.sync.tests.test_tasks.timeout_task'),
```

Comment:
I'm not a fan of adding a new library just to wait for this, and I'm still uncomfortable with having tests that are reliant on the timing between two separate areas of the code. It seems very flimsy.

I think at this point we should just mock the cache for this one test and confirm the timeout was passed properly and get on with our lives.

(You shouldn't need to move that cache import to mock it so still move it to the top of the file)